Modules/Loadable/Colors/MRMLDM/vtkMRMLColorLegendDisplayableManager.cxx

```diff
@@ -254,19 +296,34 @@ bool vtkMRMLColorLegendDisplayableManager::vtkInternal::UpdateActor(vtkMRMLColor
     vtkMRMLVolumeDisplayNode* volumeDisplayNode = vtkMRMLVolumeDisplayNode::SafeDownCast(primaryDisplayNode);
     if (volumeDisplayNode)
       {
+      vtkMRMLVolumeNode* volumeNode = vtkMRMLVolumeNode::SafeDownCast(volumeDisplayNode->GetDisplayableNode());
       // Volumes are special case, their visibility can be determined from slice view logics
       if (this->SliceCompositeNode)
         {
         // 2D view
-        visible &= this->IsVolumeVisibleInSliceView(vtkMRMLVolumeNode::SafeDownCast(volumeDisplayNode->GetDisplayableNode()));
+        visible &= this->IsVolumeVisibleInSliceView(volumeNode);
         }
       else
         {
         // 3D view
-        // For now don't show color legends for volumes in 3D views.
-        // In the future, color legend could be shown for volumes that are shown in slice views
+        // Color legend is be shown for volumes that are shown in slice views
         // that are visible in the 3D view.
-        visible = false;
+        std::vector< vtkMRMLNode* > sliceNodes;
+        this->External->GetMRMLScene()->GetNodesByClass("vtkMRMLSliceNode", sliceNodes);
```

Comment:
Since this update function can be called quite often, it is good to pay a little attention to performance.
GetNodesByClass is a relatively expensive function, so it would be better to get the slice nodes (and model display nodes, etc. from the application logic). Something like this:

```
vtkMRMLApplicationLogic* mrmlAppLogic = this->External->GetMRMLApplicationLogic();
        vtkCollection* sliceLogics = mrmlAppLogic ? mrmlAppLogic->GetSliceLogics() : nullptr;
        if (sliceLogics)
          {
          vtkMRMLSliceLogic* sliceLogic;
          vtkCollectionSimpleIterator it;
          for (sliceLogics->InitTraversal(it);
            (sliceLogic = vtkMRMLSliceLogic::SafeDownCast(sliceLogics->GetNextItemAsObject(it)));)
            {
             ...
```